interp/builtin.go

```diff
@@ -800,13 +801,95 @@ func (r *Runner) builtinCode(ctx context.Context, pos syntax.Pos, name string, a
 				return 2
 			}
 		}
+
+	case "readarray", "mapfile":
+		dropDelim := false
+		delim := "\n"
+		arrayName := "MAPFILE"
+		fp := flagParser{remaining: args}
+		for fp.more() {
+			switch flag := fp.flag(); flag {
+			case "-t":
+				// Remove the delim from each line read
+				dropDelim = true
+			case "-d":
+				if len(fp.remaining) == 0 {
+					r.errf("%s: -d: option requires an argument\n", name)
+					return 2
+				}
+				delim = fp.value()
+				if delim == "" {
+					// Bash sets the delim to an ASCII NUL if provided with an empty
+					// string.
+					delim = "\x00"
+				}
+			default:
+				r.errf("%s: invalid option %q\n", name, flag)
+				return 2
+			}
+		}
+
+		args := fp.args()
+
+		if len(args) > 1 {
+			r.errf("%s: Only one array name may be specified, %v\n", name, args)
+			return 2
+		}
+
+		if len(args) == 1 {
+			if !syntax.ValidName(args[0]) {
+				r.errf("%s: invalid identifier %q\n", name, args[0])
+				return 2
+			}
+			arrayName = args[0]
+		}
+
+		var vr expand.Variable
+		vr.Kind = expand.Indexed
+		scanner := bufio.NewScanner(r.stdin)
+		scanner.Split(mapfileSplit(delim[0], dropDelim))
+		for scanner.Scan() {
+			vr.List = append(vr.List, scanner.Text())
+		}
+		if err := scanner.Err(); err != nil {
+			r.errf("%s: unable to read, %v", name, err)
+			return 2
+		}
+		r.setVarInternal(arrayName, vr)
+
+		return 0
+
 	default:
 		// "umask", "fg", "bg",
 		panic(fmt.Sprintf("unhandled builtin: %s", name))
 	}
 	return 0
 }
 
+// Returns a suitable Split function for a bufio.Scanner, the code is mostly
```

Comment:
use the godoc style:

> // mapFileSplit returns ...

and also end with a period

Reply:
fixed, thanks